File: hackathon-2020-master/mobile-guide/lib/models/GuidePoint.dart
enum PointType {
  navigation,
  onePoint,
}

class GuidePoint {
  final String name;
  final String description;
  final String beaconUuid;
  final String imageUrl;
  final PointType pointType;
  final bool freeAccess;
  final bool isLast;

  GuidePoint({
    this.name,
    this.pointType,
    this.freeAccess,
    this.isLast,
    this.beaconUuid,
    this.imageUrl,
    this.description,
  });
}

File: hackathon-2020-master/admin-station/src/app/models/guidepoint.ts
export default class GuidePoint {
  uid: string;
  name: string;
  description: string;
  url: string;
  freeAccess: boolean;
  eventName: string;
  type: GuidePointType;
  lastBeacon: boolean;

  constructor(
    eventName: string,
    uid: string,
    name: string,
    description: string,
    url: string,
    freeAccess: boolean,
    type: GuidePointType,
    lastBeacon: boolean
  ) {
    this.eventName = eventName;
    this.uid = uid;
    this.name = name;
    this.description = description;
    this.url = url;
    this.freeAccess = freeAccess;
    this.type = type;
    this.lastBeacon = lastBeacon;
  }
}

export enum GuidePointType {
  Navigation = 0,
  Information = 1,
}
